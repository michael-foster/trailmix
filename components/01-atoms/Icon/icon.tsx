import React from "react";
import { StarIcon } from "../../00-assets/star-icon";
import { PlayIcon } from "../../00-assets/play-icon";
import { PlayButtonIcon } from "../../00-assets/play-button-icon";
import { CheckMarkIcon } from "../../00-assets/check-mark-icon";
import { ProfileIcon } from "../../00-assets/profile-icon";
import { LockIcon } from "../../00-assets/lock-icon";
import { LockShadowIcon } from "../../00-assets/lock-shadow-icon";
import { ChevronIcon } from "../../00-assets/chevron-icon";
import { ChevronIconFat } from "../../00-assets/chevron-icon-fat";
import { OutsideOIcon } from "../../00-assets/outside-o-icon";
import { CloseIcon } from "../../00-assets/close-icon";
import { InfoIcon } from "../../00-assets/info-icon";
import { RefreshIcon } from "../../00-assets/refresh-icon";

export type IconProps = {
  /** Fill color of the icon */
  color?: string;
  /** Size of the icon */
  size?:
    | "xsmall"
    | "small"
    | "small-plus"
    | "medium"
    | "large"
    | "xlarge"
    | "xxlarge";
  /** Type of icon */
  type:
    | "checkMark"
    | "chevron"
    | "chevronFat"
    | "close"
    | "info"
    | "lock"
    | "lockShadow"
    | "outsideO"
    | "play"
    | "playButton"
    | "profile"
    | "refresh"
    | "star";
  /** Additional classes to add to the icon component */
  classes?: string;
  /** Stroke color */
  strokeColor?: string;
  /** Stroke width */
  strokeWidth?: string;
};

const iconMap = {
  checkMark: CheckMarkIcon,
  chevron: ChevronIcon,
  chevronFat: ChevronIconFat,
  close: CloseIcon,
  info: InfoIcon,
  play: PlayIcon,
  playButton: PlayButtonIcon,
  profile: ProfileIcon,
  star: StarIcon,
  lock: LockIcon,
  lockShadow: LockShadowIcon,
  refresh: RefreshIcon,
  outsideO: OutsideOIcon,
};

export function Icon({
  color = "currentColor", // TODO - move this to a constants file or theme
  size,
  type,
  classes = "",
  strokeColor = "",
  strokeWidth = "",
}: IconProps) {
  const IconComponent = iconMap[type];

  return (
    <IconComponent
      classes={`${size ? `icon--${size}` : ""} ${classes}`.trim()}
      fillColor={color}
      strokeColor={strokeColor}
      strokeWidth={strokeWidth}
    />
  );
}
