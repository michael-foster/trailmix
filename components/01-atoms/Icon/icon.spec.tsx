import React from "react";
import { render } from "@testing-library/react";

import * as Icons from "./icon.composition";

describe("Icon", () => {
  test("snapshots", () => {
    const icons = Object.values(Icons).sort((a, b) =>
      a.name.localeCompare(b.name)
    );

    icons.forEach((icon) => {
      const Component = icon;

      const { asFragment } = render(<Component />);
      expect(asFragment()).toMatchSnapshot();
    });
  });
});
