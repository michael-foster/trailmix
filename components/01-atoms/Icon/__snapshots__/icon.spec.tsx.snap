// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon snapshots 1`] = `
<DocumentFragment>
  <svg
    class="icon--small"
    fill="none"
    role="img"
    viewBox="0 0 14 9"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M12.355 0.354999L7 5.69833L1.645 0.354998L6.11959e-07 2L7 9L14 2L12.355 0.354999Z"
      fill="currentColor"
      stroke=""
      stroke-width=""
    />
  </svg>
</DocumentFragment>
`;

exports[`Icon snapshots 2`] = `
<DocumentFragment>
  <svg
    class="icon--large"
    role="img"
    viewBox="0 0 516.06 515.61"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M258.03 37.328c-121.68 0-220.67 98.606-220.67 220.29s98.989 220.67 220.67 220.67 220.67-98.989 220.67-220.67-98.99-220.29-220.67-220.29zm0 22.603c109.31 0 197.68 88.369 197.68 197.68s-88.369 197.68-197.68 197.68-197.68-88.369-197.68-197.68 88.369-197.68 197.68-197.68zm0 6.896l-44.82 129.11-136.77 2.68 108.8 82.37-39.46 131.02 112.25-78.15 111.87 78.15-39.46-131.02 108.8-82.37-136.39-2.68-44.82-129.11z"
      fill="#568203"
      stroke=""
      stroke-width=""
    />
  </svg>
</DocumentFragment>
`;

exports[`Icon snapshots 3`] = `
<DocumentFragment>
  <svg
    class="icon--small"
    role="img"
    viewBox="0 0 18.01 17.98"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M9 18.3145C13.9706 18.3145 18 14.285 18 9.31445C18 4.34389 13.9706 0.314453 9 0.314453C4.02944 0.314453 0 4.34389 0 9.31445C0 14.285 4.02944 18.3145 9 18.3145Z"
      fill="currentColor"
      stroke=""
      stroke-width=""
    />
    <path
      d="M10.7041 11.7545C10.3041 12.4585 9.74411 13.2105 9.42411 13.2105C9.13611 13.2105 9.15211 12.9065 9.24811 12.5385L10.7841 6.85848L10.6881 6.76248L7.15211 7.21048L7.04011 7.72248L7.90411 7.81848C8.19211 7.83448 8.25611 8.04248 8.16011 8.44248L7.00811 12.6505C6.68811 13.8505 7.04011 14.5065 8.08011 14.5065C9.60011 14.5065 10.5121 13.3705 11.1521 12.0745L10.7041 11.7545ZM9.98411 3.59448C9.08811 3.59448 8.65611 4.26648 8.65611 4.89048C8.65611 5.53048 9.08811 5.96248 9.80811 5.96248C10.5441 5.96248 11.1201 5.49848 11.1201 4.66648C11.1201 3.96248 10.6241 3.59448 9.98411 3.59448Z"
      fill="white"
    />
  </svg>
</DocumentFragment>
`;

exports[`Icon snapshots 4`] = `
<DocumentFragment>
  <svg
    class="icon--large"
    fill="none"
    role="img"
    viewBox="0 0 34 34"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M14 24L22 17L14 10L14 24Z"
      fill="currentColor"
      stroke=""
      stroke-width=""
    />
  </svg>
</DocumentFragment>
`;

exports[`Icon snapshots 5`] = `
<DocumentFragment>
  <svg
    class="icon--small"
    fill="none"
    viewBox="0 0 16 21"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M2.91374 6.95999H3.01374V6.85999V4.89999C3.01374 3.62667 3.51506 2.40567 4.40713 1.50557C5.29917 0.605492 6.50884 0.1 7.76998 0.1C8.39446 0.1 9.01284 0.224106 9.58983 0.465255C10.1668 0.706405 10.6911 1.05989 11.1328 1.50557C11.5745 1.95125 11.925 2.48039 12.1641 3.06282C12.4032 3.64525 12.5262 4.26952 12.5262 4.89999V6.85999V6.95999H12.6262H13.5975C14.0858 6.95999 14.5544 7.15573 14.9 7.50445C15.2456 7.8532 15.44 8.32639 15.44 8.81998V18.62C15.44 19.1136 15.2456 19.5868 14.9 19.9355C14.5544 20.2842 14.0858 20.48 13.5975 20.48H1.9425C1.45413 20.48 0.985576 20.2842 0.639971 19.9355C0.294335 19.5868 0.1 19.1136 0.1 18.62V8.81998C0.1 7.78689 0.929871 6.95999 1.9425 6.95999H2.91374ZM4.75624 6.85999V6.95999H4.85624H10.6837H10.7837V6.85999V4.89999C10.7837 4.09402 10.4664 3.32089 9.90134 2.75071C9.33622 2.1805 8.56957 1.86 7.76998 1.86C6.9704 1.86 6.20375 2.1805 5.63863 2.75071C5.07354 3.32089 4.75624 4.09402 4.75624 4.89999V6.85999ZM7.76998 15.78C8.31198 15.78 8.83159 15.5627 9.21456 15.1763C9.5975 14.7899 9.81248 14.266 9.81248 13.72C9.81248 12.5775 8.89411 11.66 7.76998 11.66C7.22799 11.66 6.70838 11.8772 6.32541 12.2637C5.94246 12.65 5.72749 13.1739 5.72749 13.72C5.72749 14.266 5.94246 14.7899 6.32541 15.1763C6.70838 15.5627 7.22799 15.78 7.76998 15.78Z"
      fill="currentColor"
      stroke=""
      stroke-width=""
    />
  </svg>
</DocumentFragment>
`;

exports[`Icon snapshots 6`] = `
<DocumentFragment>
  <svg
    class="icon--small"
    fill="none"
    viewBox="0 0 17 21"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8.5278 16.0998C9.04298 16.0998 9.53706 15.8933 9.90135 15.5258C10.2656 15.1582 10.4703 14.6597 10.4703 14.1398C10.4703 13.052 9.59617 12.1798 8.5278 12.1798C8.01262 12.1798 7.51853 12.3863 7.15425 12.7539C6.78996 13.1215 6.5853 13.62 6.5853 14.1398C6.5853 14.6597 6.78996 15.1582 7.15425 15.5258C7.51853 15.8933 8.01262 16.0998 8.5278 16.0998ZM14.3553 7.27985C14.8705 7.27985 15.3645 7.48635 15.7288 7.85392C16.0931 8.22149 16.2978 8.72002 16.2978 9.23984V19.0398C16.2978 19.5596 16.0931 20.0582 15.7288 20.4258C15.3645 20.7933 14.8705 20.9998 14.3553 20.9998H2.70031C2.18513 20.9998 1.69105 20.7933 1.32676 20.4258C0.962468 20.0582 0.757813 19.5596 0.757813 19.0398V9.23984C0.757813 8.15205 1.63194 7.27985 2.70031 7.27985H3.67156V5.31985C3.67156 4.02029 4.18319 2.77396 5.09392 1.85503C6.00464 0.936108 7.23984 0.419861 8.5278 0.419861C9.16553 0.419861 9.79701 0.546603 10.3862 0.79285C10.9754 1.0391 11.5107 1.40003 11.9617 1.85503C12.4126 2.31004 12.7703 2.85021 13.0144 3.44471C13.2584 4.0392 13.384 4.67638 13.384 5.31985V7.27985H14.3553ZM8.5278 2.37986C7.75502 2.37986 7.0139 2.68961 6.46747 3.24096C5.92104 3.79232 5.61405 4.54012 5.61405 5.31985V7.27985H11.4415V5.31985C11.4415 4.54012 11.1346 3.79232 10.5881 3.24096C10.0417 2.68961 9.30057 2.37986 8.5278 2.37986Z"
      fill="#EA2127"
    />
    <path
      d="M2.91374 6.95999H3.01374V6.85999V4.89999C3.01374 3.62667 3.51506 2.40567 4.40713 1.50557C5.29917 0.605492 6.50884 0.1 7.76998 0.1C8.39446 0.1 9.01284 0.224106 9.58983 0.465255C10.1668 0.706405 10.6911 1.05989 11.1328 1.50557C11.5745 1.95125 11.925 2.48039 12.1641 3.06282C12.4032 3.64525 12.5262 4.26952 12.5262 4.89999V6.85999V6.95999H12.6262H13.5975C14.0858 6.95999 14.5544 7.15573 14.9 7.50445C15.2456 7.8532 15.44 8.32639 15.44 8.81998V18.62C15.44 19.1136 15.2456 19.5868 14.9 19.9355C14.5544 20.2842 14.0858 20.48 13.5975 20.48H1.9425C1.45413 20.48 0.985576 20.2842 0.639971 19.9355C0.294335 19.5868 0.1 19.1136 0.1 18.62V8.81998C0.1 7.78689 0.929871 6.95999 1.9425 6.95999H2.91374ZM4.75624 6.85999V6.95999H4.85624H10.6837H10.7837V6.85999V4.89999C10.7837 4.09402 10.4664 3.32089 9.90134 2.75071C9.33622 2.1805 8.56957 1.86 7.76998 1.86C6.9704 1.86 6.20375 2.1805 5.63863 2.75071C5.07354 3.32089 4.75624 4.09402 4.75624 4.89999V6.85999ZM7.76998 15.78C8.31198 15.78 8.83159 15.5627 9.21456 15.1763C9.5975 14.7899 9.81248 14.266 9.81248 13.72C9.81248 12.5775 8.89411 11.66 7.76998 11.66C7.22799 11.66 6.70838 11.8772 6.32541 12.2637C5.94246 12.65 5.72749 13.1739 5.72749 13.72C5.72749 14.266 5.94246 14.7899 6.32541 15.1763C6.70838 15.5627 7.22799 15.78 7.76998 15.78Z"
      fill="#FDD20A"
      stroke="black"
      stroke-width="0.2"
    />
  </svg>
</DocumentFragment>
`;

exports[`Icon snapshots 7`] = `
<DocumentFragment>
  <svg
    class="icon--medium"
    fill="none"
    viewBox="0 0 20 16"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M1.94483 7.70578L0 9.69527L6.2069 16L20 1.98949L18.0552 0L6.2069 12.035L1.94483 7.70578Z"
      fill="currentColor"
      stroke=""
      stroke-width=""
    />
  </svg>
</DocumentFragment>
`;

exports[`Icon snapshots 8`] = `
<DocumentFragment>
  <svg
    class="icon--small"
    fill="none"
    viewBox="0 0 179 191"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M177.168 77.7251C169.308 35.2311 138.888 0.411743 89.8094 0.411743C33.2849 0.411743 0.703125 41.9263 0.703125 96.604C0.703125 149.717 36.2311 190.828 89.8094 190.828C140.853 190.828 174.815 153.653 178.737 103.681C178.737 103.681 178.919 94.9915 178.659 90.4972C178.379 86.215 177.878 81.9536 177.168 77.7251ZM94.7298 162.298C68.0286 162.489 51.7407 119.816 51.7407 82.2403C51.7407 60.6051 59.9801 27.7566 84.5222 27.7566C117.113 27.7566 127.908 80.4725 127.908 106.041C127.902 130.241 120.05 162.298 94.7298 162.298Z"
      fill="currentColor"
      stroke=""
      stroke-width=""
    />
  </svg>
</DocumentFragment>
`;

exports[`Icon snapshots 9`] = `
<DocumentFragment>
  <svg
    class="icon--small"
    fill="none"
    role="img"
    viewBox="0 0 150 150"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M75 135C41.925 135 15 108.075 15 75C15 41.925 41.925 15 75 15C108.075 15 135 41.925 135 75C135 108.075 108.075 135 75 135ZM75 0C65.1509 0 55.3982 1.93993 46.2987 5.70904C37.1993 9.47814 28.9314 15.0026 21.967 21.967C7.90176 36.0322 0 55.1088 0 75C0 94.8912 7.90176 113.968 21.967 128.033C28.9314 134.997 37.1993 140.522 46.2987 144.291C55.3982 148.06 65.1509 150 75 150C94.8912 150 113.968 142.098 128.033 128.033C142.098 113.968 150 94.8912 150 75C150 65.1509 148.06 55.3982 144.291 46.2987C140.522 37.1993 134.997 28.9314 128.033 21.967C121.069 15.0026 112.801 9.47814 103.701 5.70904C94.6018 1.93993 84.8491 0 75 0ZM60 108.75L105 75L60 41.25V108.75Z"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`Icon snapshots 10`] = `
<DocumentFragment>
  <svg
    class="icon--small"
    fill="none"
    viewBox="0 0 14 14"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M6.5625 8C4.37813 8 0 9.17 0 11.5V14H13.125V11.5C13.125 9.17 8.74687 8 6.5625 8ZM6.5625 6C7.30842 6 8.02379 5.68393 8.55124 5.12132C9.07868 4.55871 9.375 3.79565 9.375 3C9.375 2.20435 9.07868 1.44129 8.55124 0.87868C8.02379 0.316071 7.30842 0 6.5625 0C5.81658 0 5.10121 0.316071 4.57376 0.87868C4.04632 1.44129 3.75 2.20435 3.75 3C3.75 3.79565 4.04632 4.55871 4.57376 5.12132C5.10121 5.68393 5.81658 6 6.5625 6Z"
      fill="currentColor"
      stroke=""
      stroke-width=""
    />
  </svg>
</DocumentFragment>
`;

exports[`Icon snapshots 11`] = `
<DocumentFragment>
  <svg
    class="icon--small"
    fill="none"
    role="img"
    viewBox="0 0 16 20"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M8 4V0L3 5l5 5V6a6 6 0 1 1-6 6H0a8 8 0 1 0 8-8Z"
      fill="currentColor"
    />
  </svg>
</DocumentFragment>
`;

exports[`Icon snapshots 12`] = `
<DocumentFragment>
  <svg
    class="icon--small"
    fill="none"
    role="img"
    viewBox="0 0 12 21"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M11.0881 10.8252L11.2619 10.6496L11.0883 10.4738L1.56405 0.82438L1.38636 0.64436L1.20843 0.82414L0.822311 1.21428L0.648262 1.39014L0.822311 1.566L9.81216 10.6494L0.822311 19.7327L0.642156 19.9148L0.828479 20.0905L1.2146 20.4546L1.39214 20.622L1.56381 20.4486L11.0881 10.8252Z"
      fill="currentColor"
      stroke=""
      stroke-width=""
    />
  </svg>
</DocumentFragment>
`;

exports[`Icon snapshots 13`] = `
<DocumentFragment>
  <svg
    class="icon--small"
    viewBox="0 0 14 14"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M14 1.41L12.59 0L7 5.59L1.41 0L0 1.41L5.59 7L0 12.59L1.41 14L7 8.41L12.59 14L14 12.59L8.41 7L14 1.41Z"
      fill="currentColor"
      stroke=""
      stroke-width=""
    />
  </svg>
</DocumentFragment>
`;

exports[`Icon snapshots 14`] = `
<DocumentFragment>
  <svg
    class="icon--small"
    role="img"
    viewBox="0 0 516.06 515.61"
    xmlns="http://www.w3.org/2000/svg"
  >
    <path
      d="M258.03 37.328c-121.68 0-220.67 98.606-220.67 220.29s98.989 220.67 220.67 220.67 220.67-98.989 220.67-220.67-98.99-220.29-220.67-220.29zm0 22.603c109.31 0 197.68 88.369 197.68 197.68s-88.369 197.68-197.68 197.68-197.68-88.369-197.68-197.68 88.369-197.68 197.68-197.68zm0 6.896l-44.82 129.11-136.77 2.68 108.8 82.37-39.46 131.02 112.25-78.15 111.87 78.15-39.46-131.02 108.8-82.37-136.39-2.68-44.82-129.11z"
      fill="currentColor"
      stroke=""
      stroke-width=""
    />
  </svg>
</DocumentFragment>
`;
